jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticProperties.java

```diff
@@ -254,16 +267,52 @@ public static void dumpPropertiesIfDebug(String msg, final Properties props) {
 	 * @return String representation of props
 	 */
 	public static String printProperties(final Properties props) {
-		final StringWriter buff = new StringWriter(1000);
+		StringWriter buff = new StringWriter(1000);
 		try (PrintWriter buffWriter = new PrintWriter(buff)) {
 			props.entrySet().stream().sorted(Comparator.comparing(e -> e.getKey().toString())).forEach(theEntry -> {
 				buffWriter.print(theEntry.getKey());
 				buffWriter.print("="); //$NON-NLS-1$
 				buffWriter.println((String) theEntry.getValue());
 			});
 		}
-		final String result = buff.toString();
-		return result;
+		return buff.toString();
+	}
+	
+	/**
+	 * Print the result string of a command that produces a single string. Print an
+	 * error message if the command resulted in an error.
+	 * 
+	 * @return String result or error message
+	 */
+	public String printStringResult() {
+		StringWriter buff = new StringWriter(1000);
+		try (PrintWriter buffWriter = new PrintWriter(buff)) {
+			if (Boolean.parseBoolean(getPropertyOrNull(IPC.PROPERTY_DIAGNOSTICS_ERROR))) {
+				buffWriter.printf("Error: %s%n", getString(IPC.PROPERTY_DIAGNOSTICS_ERRORTYPE)); //$NON-NLS-1$
+				String msg = getPropertyOrNull(IPC.PROPERTY_DIAGNOSTICS_ERRORMSG);
+				if (null != msg) {
+					buffWriter.println(msg);
+				}
+			} else {
+				buffWriter.println(getString(DiagnosticsInfo.DIAGNOSTICS_STRING_RESULT));
+			}
+		} catch (IOException e) {
+			buff = new StringWriter();
+			buff.append("Error parsing properties: " + e.toString() + System.lineSeparator()); //$NON-NLS-1$
```

Comment:
This would be better as three calls to `append` rather than the implicit use of `StringBuilder`.